src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyNameAttribute.cs

```diff
@@ -23,6 +23,6 @@ public JsonPropertyNameAttribute(string propertyName)
         /// <summary>
         /// The name of the property.
         /// </summary>
-        public string Name { get; set; }
+        public string Name { get; private set; }
```

Comment:
you can skip the setter here i.e. ```public string Name { get; }``` will be readonly property. we wont need a private setter as ctor is the only place we are setting the Name property